tinygrad/llops/ops_gpu.py

```diff
@@ -325,7 +325,7 @@ def codegen(self) -> Callable:
 
     # kernel function definition
     function_name = ("re_S" if self.reduceop else "ew_S") + '_'.join([str(x) for x in self.bufs[0].shape if x != 1])
-    buftypes = [f"{'read_only' if i > 0 else 'write_only'} image2d_t" if isinstance(x._buf, CLImage) else (CLProgram.buffer_prefix+self.buftokens[i].decltype()) for i,x in enumerate(self.bufs)]
+    buftypes = [f"{'read_only' if i > 0 else 'write_only'} image2d_t" if isinstance(x._buf, CLImage) else (CLProgram.buffer_prefix+self.buftokens[i].decltype() + "restrict" if CLANG else "") for i,x in enumerate(self.bufs)]
```

Comment:
You need to wrap `if` with parenthesis

Reply:
thanks for the catch + explanation, fixed in a new commit, I can combine them if wanted.